### PR TITLE
Move requirements to requirements.txt

### PR DESCRIPTION
**Pull request by jeremycline**
This makes it easy to `pip install -r` or use Ansible to install from a
requirements file.

Signed-off-by: Jeremy Cline jeremy@jcline.org

### CODE REVIEW

**Comment by jeremycline** [comment]
The change in `.travis.yml` is due to Travis hitting https://github.com/shazow/urllib3/issues/986. It had setuptools 12.0.5 which is apparently too old.

**Comment by ashcrow** [comment]
@jeremycline The error cropping up is due to fmn.hinting not being available.

```
    from fmn.lib.hinting import hint, prefixed as _
```

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ python:
  - "3.3"
 
 install:
+ - pip install --upgrade setuptools
  - python setup.py install
  - pip install nose mock sqlalchemy unittest2 fmn.lib
```
Comment (ashcrow) [inline .travis.yml]: nit: Why not move these to `tests-requirements.txt` as well?

**Comment by ashcrow** [comment]
Overall I think this change makes sense and makes it easier to work with the package.

**Comment by jeremycline** [comment]
I'm not entirely sure what the deal was with importing `fmn.lib` stuff, but I can't seem to reproduce it and the tests are now passing after a few additional tweaks (like requiring `html5lib` for the tests).

File: .travis.yml (hunk 1 above)
Comment (jeremycline) [reply .travis.yml]: I took care of this - a lot of these don't seem to be actual test requirements (or are pulled in by other libraries that _do_ depend on them).

**Review by ashcrow — APPROVED** [review]
LGTM!

File: setup.py [hunk 2]
```diff
@@ -19,17 +19,24 @@ def get_description():
...
+    :type  requirements_file: str
+
+    :returns: the list of requirements
+    :return type: list
+    """
+    with open(filename) as f:
+        return [
+            line.rstrip().split('#')[0]
+            for line in f.readlines()
+            if not line.startswith('#')
```
Comment (bowlofeggs) [inline setup.py:37]: Getting fancy there allowing for basic commenting, eh?

File: setup.py [hunk 3]
```diff
@@ -19,17 +19,24 @@ def get_description():
...
-tests_require = [
-    'nose',
-]
+
+def get_requirements(filename='requirements.txt'):
+    """
+    Get the contents of a file listing the requirements.
+
+    :param requirements_file: path to a requirements file
+    :type  requirements_file: str
```
Comment (bowlofeggs) [inline setup.py:28]: The parameter name is filename, not requirements_file.

File: setup.py [hunk 4]
```diff
@@ -41,8 +48,8 @@ def get_description():
     url="https://github.com/fedora-infra/fmn.rules",
     download_url="https://pypi.python.org/pypi/fmn.rules/",
     license='LGPLv2+',
-    install_requires=requires,
-    tests_require=tests_require,
+    install_requires=get_requirements('requirements.txt'),
+    tests_require=get_requirements('tests-requirements.txt'),
```
Comment (bowlofeggs) [inline setup.py:52]: nitpick that you can ignore, but I think "test-requirements" sounds more natural than "tests-requirements".

**Review by bowlofeggs — APPROVED** [review]
Once you fix that docblock's parameter names to match the real name, LGTM! I don't need to re-review, so feel free to merge once that is fixed.

File: setup.py (hunk 2 above)
Comment (jeremycline) [reply setup.py:37]: It is a little, but I wanted it to behave like other projects (like https://pagure.io/pagure/blob/master/f/setup.py) so people wouldn't try something they can do elsewhere and be surprised.

**Pull request merged**